src/api/utils.py

```diff
@@ -28,5 +28,23 @@ def failure_response_with_query(query, message, status=status.HTTP_404_NOT_FOUND
 
 def modify_attribute(model, attr_name, attr_value):
     """Modify an attribute if it isn't None and has been changed."""
-    if attr_value is not None and attr_value != getattr(model, attr_name):
+    if (
+        attr_value is not None
+        and attr_value != "null"
+        and attr_value != getattr(model, attr_name)
+    ):
         setattr(model, attr_name, attr_value)
+
+
+def update_many_to_many_set(class_name, existing_set, ids):
+    """Update a ManyToMany relation set for a person object (i.e. Locations, Groups, etc)"""
+    if ids is not None:
+        new_objects = []
+        for id in ids:
+            new_object = class_name.objects.filter(id=id)
+            if not new_object:
+                return failure_response(
+                    f"{class_name.__name__} id {id} does not exist."
+                )
+            new_objects.append(new_object[0])
+        existing_set.set(new_objects)
```

Comment:
should always check error case/none case and return immediately first for readability